### PR TITLE
feat: handle when user anonymous not found

### PR DESCRIPTION
**Pull request by n-permana**

### CODE REVIEW

**Comment by bastian-purrer** [comment]
Image description   CodeRabbit



Currently reviewing new changes in this PR...

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between 75f5c16ca43b4566215a2aa2b4ca1dbd86865406 and 48cbe53c63cb3ae276db53203d24d1469e508b52 commits.
</details>

<details>
<summary>Files selected (1)</summary>

* src/processes/scoring/calc-karma-score.js (1)
</details>






---

**Review by bastian-purrer — COMMENTED** [review]
Image description   CodeRabbit

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between 75f5c16ca43b4566215a2aa2b4ca1dbd86865406 and 48cbe53c63cb3ae276db53203d24d1469e508b52 commits.
</details>

<details>
<summary>Files selected (1)</summary>

* src/processes/scoring/calc-karma-score.js (1)
</details>





<details>
<summary>Files not summarized due to errors (1)</summary>

* src/processes/scoring/calc-karma-score.js (nothing obtained from openai)

</details>


<details>
<summary>Files not reviewed due to errors (1)</summary>

* src/processes/scoring/calc-karma-score.js (no response)

</details>


<details>
<summary>Review comments generated (0)</summary>

* Review: 0
* LGTM: 0

</details>

---

<details>
<summary>Tips</summary>

### Chat with Image description  CodeRabbit Bot (`@coderabbitai`)
- Reply on review comments left by this bot to ask follow-up questions. A review comment is a comment on a diff or a file.
- Invite the bot into a review comment chain by tagging `@coderabbitai` in a reply.

### Code suggestions
- The bot may make code suggestions, but please review them carefully before committing since the line number ranges may be misaligned. 
- You can edit the comment made by the bot and manually tweak the suggestion if it is slightly off.

### Pausing incremental reviews
- Add `@coderabbitai: ignore` anywhere in the PR description to pause further reviews from the bot.

</details>

File: src/processes/scoring/calc-karma-score.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ const calculateP = (p) => {
 };
 
 const getUserScore = (user) => {
-  return typeof user.user_score === 'number' ? user.user_score : 0;
+  return typeof user?.user_score === 'number' && user ? user.user_score : 0;
```
Comment (fajarism) [inline src/processes/scoring/calc-karma-score.js:15]: for better readability, please change the order to
```
user && user?.user_score === 'number'
```

**Review by bastian-purrer — COMMENTED** [review]
Image description   CodeRabbit

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between 48cbe53c63cb3ae276db53203d24d1469e508b52 and 8f3f94b9e708aa07029e5fa51f675d39847d4470 commits.
</details>

<details>
<summary>Files selected (1)</summary>

* src/processes/scoring/calc-karma-score.js (1)
</details>





<details>
<summary>Files not summarized due to errors (1)</summary>

* src/processes/scoring/calc-karma-score.js (nothing obtained from openai)

</details>


<details>
<summary>Files not reviewed due to errors (1)</summary>

* src/processes/scoring/calc-karma-score.js (no response)

</details>


<details>
<summary>Review comments generated (0)</summary>

* Review: 0
* LGTM: 0

</details>

---

<details>
<summary>Tips</summary>

### Chat with Image description  CodeRabbit Bot (`@coderabbitai`)
- Reply on review comments left by this bot to ask follow-up questions. A review comment is a comment on a diff or a file.
- Invite the bot into a review comment chain by tagging `@coderabbitai` in a reply.

### Code suggestions
- The bot may make code suggestions, but please review them carefully before committing since the line number ranges may be misaligned. 
- You can edit the comment made by the bot and manually tweak the suggestion if it is slightly off.

### Pausing incremental reviews
- Add `@coderabbitai: ignore` anywhere in the PR description to pause further reviews from the bot.

</details>

**Review by brm-stnd — APPROVED** [review]

**Pull request merged**